File: src/update/pubkey.ts
/**
 * ⚠️ SHARED SOURCE OF TRUTH — do NOT delete with the rest of the legacy client.
 *
 * The rest of the Bun/TypeScript self-updater has been removed; the shipping
 * updater lives in the Rust crate `rust/climon-update`. This file is retained
 * because `rust/climon-update/build.rs` reads `UPDATE_PUBLIC_KEY_B64` from here
 * at build time so the Rust updater embeds the exact same base64 Ed25519 public
 * key. Deleting this file breaks the Rust client build. `scripts/gen-update-keys.ts`
 * also documents this file as the place to store the public key.
 */
/**
 * Base64 of the 32-byte raw Ed25519 public key used to verify update artifacts.
 * REPLACE this placeholder with the real public key from `scripts/gen-update-keys.ts`
 * before shipping signed releases. An empty value disables verification-backed
 * updates (the updater refuses to apply unverifiable downloads).
 */
export const UPDATE_PUBLIC_KEY_B64 =
  "MTrrT1WssqQuGrVEBPqZWKqPifHo2xZXgByXLAX0l/o=";

File: rust/climon-update/src/pubkey.rs
//! Embedded Ed25519 update public key. Port of `src/update/pubkey.ts`.
//!
//! The base64 value is injected at build time from `src/update/pubkey.ts` by
//! `build.rs` so the Rust updater verifies against the *exact same* key as the
//! Bun client. An empty value disables verification-backed updates.

/// Base64 of the 32-byte raw Ed25519 public key used to verify update artifacts.
/// Synced from `src/update/pubkey.ts` at build time (see `build.rs`).
pub const UPDATE_PUBLIC_KEY_B64: &str = env!("CLIMON_UPDATE_PUBKEY_B64");

#[cfg(test)]
mod tests {
    use super::*;
    use base64::{engine::general_purpose::STANDARD, Engine};

    #[test]
    fn is_a_string() {
        // Mirrors the TS "is a string (may be empty until provisioned)" test.
        let _: &str = UPDATE_PUBLIC_KEY_B64;
    }

    #[test]
    fn when_set_decodes_to_32_raw_bytes() {
        if UPDATE_PUBLIC_KEY_B64.is_empty() {
            return;
        }
        let bytes = STANDARD.decode(UPDATE_PUBLIC_KEY_B64).unwrap();
        assert_eq!(bytes.len(), 32);
    }
}
